### DISCUSSION TITLE
Workflow to queue videos in full window?

### DISCUSSION THREAD

**Discussion by DejayRezme** [FreeTubeApp/FreeTube#4456] [General]
Hello! I just discovered freetube so I wondered if this is already supported somehow or planned or if this should be a new feature suggestion:

I've "developed" this "workflow" for youtube website to watch videos somewhat similar to how I would watch a TV somewhat in the background. Basically I'd like to have a video playing in "full window" and being able to easily add videos to queue up. In youtube you could make a "watch later" playlist and then watch those (e.g. a number news videos from many subscribed channels), but you needed to finish the list of videos first and could not add to it later.

It would be awesome if the full window had a "pop up" when you move to the right window edge that shows other recent videos *from your subscription list*. I'm generally not that interested in trending or youtube's profit oriented suggestions.

I have my 32" 4k monitor split into 2x2 grid with the left half often occupied by a web browser, the top right with a youtube window or a video play with a tv show, and the top right with some other utility app. For this freetube is already amazing because it can be more compact. But as, when the video is finished (or 90% finished but still playing) you have to click full window, click on subscription, scroll to where the last video was, click the next video you want to watch, then click full window again.

- Pop up list for videos from your subscriptions / peek (date sorted and jump to current videos date)
- A completely undecorated window for "full window" without a bar just like MPV
- Mouse wheel in full screen / full window should forward / rewind without having to have the video in foreground. As is I have to click on freetube, click again to resume, press a key, then go back. *EDIT: already works if enabled in settings lol*
- Middle mouse click on a video in subscriptions should add it to "watch later". And/or shift-left click
- Option to open videos or new windows with "default to full window" (theater mode?)
- Play all new videos in your subscriptions in chronological order

Freetube looks amazing so far, thank you so much!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> * Pop up list for videos from your subscriptions / peek (date sorted and jump to current videos date)

This wont be needed when https://github.com/FreeTubeApp/FreeTube/issues/547 is implemented because then you can just queue up videos

> * A completely undecorated window for "full window" without a bar just like MPV

You can acomplish this by hitting F11 after enabling fullwindow but watch out f11 causes some weird behavior at the moment see https://github.com/FreeTubeApp/FreeTube/issues/3220

> * Mouse wheel in full screen / full window should forward / rewind without having to have the video in foreground. As is I have to click on freetube, click again to resume, press a key, then go back.

Not sure if we can/should implement something like this. Simple example you have another application open and use scroll wheel, it could rewind/forward video but cant scroll current open application?

> * Middle mouse click on a video in subscriptions should add it to "watch later". And/or shift-left click

Watch later will be implemented after #4234 and is documented in https://github.com/FreeTubeApp/FreeTube/issues/547

> * Option to open videos or new windows with "default to full window" (theater mode?)

Documented in https://github.com/FreeTubeApp/FreeTube/issues/1047 and https://github.com/FreeTubeApp/FreeTube/issues/2295

> * Play all new videos in your subscriptions in chronological order

This one is documented https://github.com/FreeTubeApp/FreeTube/issues/413

**Reply by DejayRezme** [reply]
Thanks for taking time for a reply, I'll check out the issues you linked.

What I mean with undecorated was to have a window that has no title bar so that a video can play in a 1920x1080 large window without wasting space. MPV does this so the window controls ( like minimize and close) are only displayed on hover. I think this is accomplished by "const win = new BrowserWindow({ titleBarStyle: 'hidden' })". You can still designate the top area as your area to drag the window around. I've been meaning to learn how to build apps with electronjs so maybe I will give it a shot.

Maybe this is way too niche for my 2x2 grid setup :) But I don't like wasting the title bar since it's almost never needed. 

The mouse wheel would only work when hovering over the window. That is the default behavior for scroll events for windows app, the only change would be that mouse wheel in full window / full screen would change to forward/rewind. Currently it does nothing.

**Reply by absidue** [reply]
For the scrolling to seek the existing "Skip by Scrolling Over Video Player" player setting is probably what you are looking for.

As for that fake full screen window, that is definitely really niche and probably not something that would be worth us implementing. Hiding the title bar is one thing, but because you have hidden it, you then need to implement a completely custom one as a replacement, which not only won't look native but also leads to a worse user experience (cough discord).

If you just want a floating window with the video, you can turn on picture-in-picture, yes the controls are limited, you can thank Google for that.

**Reply by DejayRezme** [reply]
Thanks, found the skip by scrolling and it's perfect! I'm really liking freetube. And yeah I guess if I want windows without title bars I should finally switch to linux and look into customizing a window manager. 

I guess my main issue is that the video can't be as wide as the window outside of full window. With browser and YoutubeEnhancer you can do that. I guess my minimum proposal would be this:

* In theater mode do not have margins to the left and right of the player video and have the sidebar and top bar do something like "auto hide" depending on mouse proximity. 
* Add videos from a more configurable "up next" below the window to the "now playing" queue. Like with middle mouse button you "add to now playing", with left you immediately start playing.
* Have the "up next" panel show the list / grid you showed like subscriptions, search results or trending, depending on what you'd get if you click "back". Maybe with a tab view.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
These proposals are based on the fact that you are familiar with an browser extension for YT. 

Yes we try to mimic YT to make the UX somewhat familiar but that doesnt mean you should treat FT as a browser window. FT is an application and does stuff in its own way to make it usable for everyone. 

Your proposals just dont make sense and would divert from the familiarity we try to mimic from stock YT
